### PR TITLE
DragControls: Refactor API.

### PR DESCRIPTION
**Pull request by Mugen87**
Related issue: #29072, #20575

**Description**

This PR refactors the API of `DragControls`.

- `THREE.Controls` is introduced which defines a basic control interface. `DragControls` is now derived from this class.
- Side effects in the ctor are moved into a `connect()` method.
- `domElement` is now optional. If required, an instance of `DragControls` can now be created without a DOM element. When the element is set at a later point, a manual call of `connect()` configures the event listeners.

### CODE REVIEW

**Comment by puxiao** [comment]
Do you think it is appropriate to add saveState(), setState(), and reset() to Controls?

**Comment by Mugen87** [comment]
Not all controls support these methods so I would revisit this after all classes has been updated. 

If we add these methods to the `Controls` interface, we should make sure all derived classes implement them.

**Comment by mrdoob** [comment]
I think we should move `Controls.js` to `src`. Maybe `src/extras`?

**Comment by Mugen87** [comment]
Sounds good!

File: examples/jsm/controls/Controls.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+		this.state = - 1;
+
+		this.keys = {};
+		this.mouseButtons = { LEFT: null, MIDDLE: null, RIGHT: null };
+		this.touches = { ONE: null, TWO: null };
+
+	}
+
+	connect() {}
```
Comment (mrdoob) [inline examples/jsm/controls/Controls.js:22]: @Mugen87 What do you think about being able to pass the dom element here?

That way the code could look like this:

```js
const controls = new OrbitControls( camera );
controls.connect( renderer.domElement );
```

I guess we'll need to do a bit of management though:

```js
connect( element ) {

    if ( this.domElement !== element ) {

        this.disconnect();

    }

    this.domElement = element;

}
```

File: examples/jsm/controls/Controls.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { EventDispatcher } from 'three';
+
+class Controls extends EventDispatcher {
+
+	constructor( object, domElement ) {
```
Comment (mrdoob) [inline examples/jsm/controls/Controls.js:5]: @Mugen87 

`constructor( object, domElement = null )` maybe?

Right now it's `undefined` by default but subclasses set it to `null`...

File: examples/jsm/controls/Controls.js (hunk 1 above)
Comment (Mugen87) [reply examples/jsm/controls/Controls.js:22]: As a convenient feature, I guess it's fine.

That said, `domElement` is public so devs can basically achieve the same without further changes to the control classes. However, additional management code in `connect()` could prevent mistakes when devs want to exchange the DOM element (not sure how common this use case is though).

**Pull request merged**